### DISCUSSION TITLE
MSIX Dependencies vs. Windows Package Manager Dependencies

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#2153] [General]
All,

We've had a bit of confusion regarding dependencies and how they are supported in the Windows Package Manager compared with MSIX dependencies.

The MSIX manifest has a mechanism for identifying dependencies. These dependencies are not required to be explicitly declared in their corresponding Windows Package Manager Manifest. If the MSIX package is coming from the Microsoft Store, a mechanism exists to deliver the dependencies directly through the Microsoft Store delivery system. If these dependencies are not declared in the Windows Package Manager manifest, they will not be exposed to the winget.exe client.

The experimental work we have done to date (as of May 11th, 2022) will only display dependencies declared in a manifest. Some examples of these are mentioned in the migration effort the .NET team is [implementing](https://github.com/dotnet/core/issues/7428). Only those dependencies declared in a Windows Package Manager manifest will be acted on directly by the winget.exe client.

Related to:
* #163

**Comment by Kein** [comment]
So winget will never display any kind of information about MSIX dependencies, because it is not [one of] it[s] purposes as a package manager tool. Did I understand this correctly?

**Reply by denelon** [reply]
We've got some work to evaluate MSIX packages and make sure the data in the Windows Package Manager manifest is properly stated (Package Family Name, Version, Minimum OS Version, etc.). This is primarily related to our validation pipelines.

We could consider additional keys in the manifest to expose the MSIX framework dependencies, or other dependencies in the MSIX manifest, but I'd be cautious about whether the Windows Package Manager should pull in (evaluate/install) dependencies declared inside of the MSIX manifest. I'd like to understand the scenario you are looking to resolve. 

Maybe it is something we should consider. The main challenge is we don't want to have to download the MSIX just to display dependency information.

**Reply by denelon** [reply]
Also, with the advent of REST sources, we don't want to impose the same types of requirements we have in the [Windows Package Manager Community Repository](https://github.com/microsoft/winget-pkgs).

**Reply by Kein** [reply]
> The main challenge is we don't want to have to download the MSIX just to display dependency information.

Are you sure this is the only way to query dependencies from MS store for MSIX package?

**Reply by denelon** [reply]
I'm not aware of any documented interfaces we could otherwise leverage.

**Reply by denelon** [reply]
What is the scenario you are thinking of where you want to get the dependency information from an MSIX package in the Microsoft Store? It might help if I fully understood the use case.

**Reply by Kein** [reply]
Being able to install package X but not being able to see package X dependencies in package manager is just sounds wild to me. May be we can use different term than `package manager` since the subject does not fit the well-accepted criteria.
